spec/utility/circular_buffer_spec.rb

```diff
@@ -0,0 +1,91 @@
+# encoding: utf-8
+
+require 'spec_helper'
+
+describe Bugsnag::Utility::CircularBuffer do
+  describe "buffer size" do
+    it "should be 25 by default" do
+      buffer = Bugsnag::Utility::CircularBuffer.new
+
+      expect(buffer.max_items).to equal 25
+    end
+
+    it "can be set during initialization" do
+      buffer = Bugsnag::Utility::CircularBuffer.new(10)
+
+      expect(buffer.max_items).to equal 10
+    end
+
+    it "can be changed using max_items =" do
+      buffer = Bugsnag::Utility::CircularBuffer.new(10)
+      buffer.max_items = 17
+
+      expect(buffer.max_items).to equal(17)
+    end
+  end
+
+  describe "buffer items" do
+    it "initializes empty <<" do
+      buffer = Bugsnag::Utility::CircularBuffer.new
+
+      expect(buffer.to_a).to eq([])
+    end
+
+    it "can be added to with <<" do
```

Comment:
Lines 34-60 should probably be in a `describe '#<<'` block instead (and maybe lines 62-74 in `describe '#each'`, and lines 76-89 in the same `describe '#max_items='` described by a previous comment)